solana-programs/anchor/audius-data/tests/audius-data.ts

```diff
@@ -60,117 +63,96 @@ describe('audius-data', () => {
+      throw new Error(
+        `Invalid eth address - expected ${testEthAddr.toLowerCase()}, found ${returnedHex}`
+      );
     }
-
     if (!DefaultPubkey.equals(returnedSolFromChain)) {
-      throw new Error(`Unexpected public key found`)
+      throw new Error(`Unexpected public key found`);
     }
-
-    await confirmLogInTransaction(tx, metadata)
-  }
+    await confirmLogInTransaction(tx, metadata);
+  };
 
   const testInitUserSolPubkey = async ({
     message,
     pkString,
     privKey,
     newUserKey,
-    newUserAcctPDA
+    newUserAcctPDA,
   }) => {
-    const signedBytes = signBytes(Buffer.from(message), pkString)
-    const { signature, recoveryId } = signedBytes
-
-    // Get the public key in a compressed format
-    const ethPubkey = secp256k1.publicKeyCreate(privKey, false).slice(1)
-    const secpTransactionInstruction = Secp256k1Program.createInstructionWithPublicKey({
-      publicKey: Buffer.from(ethPubkey),
-      message: Buffer.from(message),
-      signature,
-      recoveryId
-    })
-
-    await provider.send(
-      (() => {
-        const tx = new Transaction();
-        tx.add(secpTransactionInstruction),
-        tx.add(
-          program.instruction.initUserSol(
-            newUserKey.publicKey,
-            {
-              accounts:
-              {
-                user: newUserAcctPDA,
-                sysvarProgram: SystemSysVarProgramKey
-              }
-            }
-          )
-        )
-        return tx;
-      })(),
-      [
-        // Signers
-      ]
-    );
-    let userDataFromChain = await program.account.user.fetch(newUserAcctPDA)
+    let initUserTx = await initUserSolPubkey({
+      provider,
+      program,
+      privateKey: pkString,
+      message,
+      userSolPubkey: newUserKey.publicKey,
+      userStgAccount: newUserAcctPDA,
+    });
+
+    let userDataFromChain = await program.account.user.fetch(newUserAcctPDA);
     if (!newUserKey.publicKey.equals(userDataFromChain.authority)) {
-      throw new Error('Unexpected public key found')
+      throw new Error("Unexpected public key found");
     }
-  }
+    let txInfo = await getTransaction(provider, initUserTx);
+    let fee = txInfo["meta"]["fee"];
+    console.log(`initUser tx = ${initUserTx} fee = ${fee}`);
+  };
 
   const testCreateTrack = async ({
     trackMetadata,
     newTrackKeypair,
     userAuthorityKey,
-    trackOwnerPDA
+    trackOwnerPDA,
+    adminStgKeypair,
   }) => {
-    let tx = await program.rpc.createTrack(
-      trackMetadata,
-      {
-        accounts: {
-          track: newTrackKeypair.publicKey,
-          user: trackOwnerPDA,
-          authority: userAuthorityKey.publicKey,
-          payer: provider.wallet.publicKey,
-          systemProgram: SystemProgram.programId
-        },
-        signers: [userAuthorityKey, newTrackKeypair]
-      }
-    )
-
-    await confirmLogInTransaction(tx, trackMetadata)
-  }
+    let trackId = await program.account.audiusAdmin.fetch(
```

Comment:
is this used? if so, using the adminStgKeypair.publicKey and calling variable trackId seems incorrect

Reply:
it actually is correct - this is the global nonce value stored in the admin account